state/dbstore.go

```diff
@@ -34,7 +35,7 @@ type Store interface {
 	Get(key string, i interface{}) (err error)
 	Put(key string, i interface{}) (err error)
 	Delete(key string) (err error)
-	Keys() (keys []string, err error)
+	Keys(prefix string) (keys []string, err error)
```

Comment:
The idea behind this function is to have a way to filter store keys by prefix, so as to have an easier time accessing just the cheques or just the balances.

Adding a `key, value` function sounds like a good idea. This seems like an optimization though, as the point of this PR was to fix the mixup between the cheques keys and the balance keys. Do you think it's better to do this now rather than in a separate issue?

Also: I'm not sure about exposing the `leveldb`: isn't hiding the underlying db one of the reasons to use the `DBStore` struct in the first place?

Reply:
I agree with @mortelli that `StateStore` is indeed an abstraction, and exposing the `LevelDB` interface, while a valid option, seems rather "breaking" that design.

Reply:
Well, I did not intend for you to expose the actual leveldb method, but to create the correct abstraction that would provide with a proper iterator as in `shed.Index.Iterate()` - which does not expose leveldb per se

Reply:
We will address this as part of https://github.com/ethersphere/swarm/pull/1554